gradle/gradle_properties.go

```diff
@@ -0,0 +1,42 @@
+package gradle
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/buildpacks/libcnb"
+)
+
+type PropertiesFile struct {
+	Binding    libcnb.Binding
+	GradleHome string
+}
+
+func (p PropertiesFile) Contribute(layer libcnb.Layer) (libcnb.Layer, error) {
+	path, ok := p.Binding.SecretFilePath("gradle.properties")
+	if !ok {
+		return layer, nil
+	}
+
+	gradlePropertiesPath := filepath.Join(p.GradleHome, "gradle.properties")
+	if err := os.Symlink(path, gradlePropertiesPath); os.IsExist(err) {
+		err = os.Remove(gradlePropertiesPath)
+		if err != nil {
+			return layer, fmt.Errorf("unable to remove old symlink for gradle.properties\n%w", err)
```

Comment:
```suggestion
			return libcnb.Layer{}, fmt.Errorf("unable to remove old symlink for gradle.properties\n%w", err)
```
It's common practice to return empty values for other return values when returning an error.

lines 31, and 34 as well.

Reply:
Fixed.